pages/index.vue

```diff
@@ -105,67 +111,95 @@
 
 <script lang>
 import { defineComponent } from '@nuxtjs/composition-api'
+import auth from '../middleware/auth'
 import addTask from '~/components/addTask.vue'
-
 export default defineComponent({
   components: { addTask },
+  middleware: auth,
   data() {
     return {
-      hello: 'hello world!',
-      todos: [
-        {
-          title: 'Henlo',
-          id: 1,
-          editing: false,
-        },
-        {
-          title: 'Frens',
-          id: 2,
-          editing: false,
-        },
-      ],
-      loading: false,
+      todos: [],
+      loading: true,
+      editTitle: '',
     }
   },
   mounted() {
-    this.getTasks()
+    this.getTasks(0)
+    this.name()
   },
   methods: {
-    async getTasks() {
-      /***
-       * @todo Fetch the tasks created by the user and display them.
-       * @todo also the function to display a single new task added
-       * @hints use store and set loading true
-       * @caution you have to assign new value to todos for it to update
-       */
+    async getTasks(i) {
+      this.$axios({
+          headers: { Authorization: 'Token ' + this.$store.getters.token },
+          url: 'https://todo-app-csoc.herokuapp.com/todo/',
+          method: 'get',
+        })
+        .then((response) => {
+          this.todos=[]
+          response.data.forEach((value)=>{
+          value.editing=false
+          this.todos.push(value)
+          })
+          if(i){this.$toast.success("Task Added!..")}
+          else{this.$toast.success("All tasks loded!..")}
```

Comment:
you could have added this toast in `addTask` element so that code would not need an additional and somewhat unnecessary parameter.